### ISSUE TITLE
SPFx Error with Dialogs in IE11

### ISSUE THREAD

**Issue by henningeiben** [SharePoint/sp-dev-docs#5331]
## Category

- [x] Bug

## Describe the bug

The `then()` after showing a dialog is not being executed; the develop-console in IE11 shows an error.

```
NotFoundError
   {
      [functions]: ,
      @@: undefined,
      @@1: undefined,
      @@2: undefined,
      @@3: undefined,
      @@4: undefined,
      @@__REACT_DND_CONTEXT_INSTANCE__: undefined,
      @@_latestValue: undefined,
      @@_state: undefined,
      @@hasInstance: undefined,
      @@isConcatSpreadable: undefined,
      @@iterator: undefined,
      @@match: undefined,
      @@observable: undefined,
      @@react.async_mode: undefined,
      @@react.concurrent_mode: undefined,
      @@react.context: undefined,
      @@react.element: undefined,
      @@react.forward_ref: undefined,
      @@react.fragment: undefined,
      @@react.fundamental: undefined,
      @@react.lazy: undefined,
      @@react.memo: undefined,
      @@react.portal: undefined,
      @@react.profiler: undefined,
      @@react.provider: undefined,
      @@react.responder: undefined,
      @@react.scope: undefined,
      @@react.strict_mode: undefined,
      @@react.suspense: undefined,
      @@react.suspense_list: undefined,
      @@replace: undefined,
      @@search: undefined,
      @@species: undefined,
      @@split: undefined,
      @@Symbol._hidden: undefined,
      @@Symbol.asyncIterator: undefined,
      @@test symbol: undefined,
      @@test symbol1: undefined,
      @@toPrimitive: undefined,
      @@toStringTag: undefined,
      @@unscopables: undefined,
      __proto__: { },
      ABORT_ERR: 20,
      code: 8,
      constructor: { },
      DATA_CLONE_ERR: 25,
      DOMSTRING_SIZE_ERR: 2,
      HIERARCHY_REQUEST_ERR: 3,
      INDEX_SIZE_ERR: 1,
      INUSE_ATTRIBUTE_ERR: 10,
      INVALID_ACCESS_ERR: 15,
      INVALID_CHARACTER_ERR: 5,
      INVALID_MODIFICATION_ERR: 13,
      INVALID_NODE_TYPE_ERR: 24,
      INVALID_STATE_ERR: 11,
      message: "NotFoundError",
      name: "NotFoundError",
      NAMESPACE_ERR: 14,
      NETWORK_ERR: 19,
      NO_DATA_ALLOWED_ERR: 6,
      NO_MODIFICATION_ALLOWED_ERR: 7,
      NOT_FOUND_ERR: 8,
      NOT_SUPPORTED_ERR: 9,
      PARSE_ERR: 81,
      QUOTA_EXCEEDED_ERR: 22,
      SECURITY_ERR: 18,
      SERIALIZE_ERR: 82,
      SYNTAX_ERR: 12,
      TIMEOUT_ERR: 23,
      TYPE_MISMATCH_ERR: 17,
      URL_MISMATCH_ERR: 21,
      VALIDATION_ERR: 16,
      WRONG_DOCUMENT_ERR: 4
   }
```

## Steps to reproduce

I have create a custom dialog extending the `BaseDialog` from `@microsoft/sp-dialog`. This dialog is shown from within a commandbar-set extension like this

```typescript
var configSelectionDialog = new CustomCommandSelectionDialog();
configSelectionDialog.show().then(_ => {
  // ...
});
```

## Expected behavior

The resolved-promise `then()` should be executed after showing a dialog.

## Developer environment

- OS: Windows 10
- Browser: IE v11
- Tooling: SPFx v1.9.1, office-ui-fabric-react 6.189.2, @pnp/polyfill-ie11 1.0.2, @pnp/sp 1.3.7

### Additional context

This seems to be the same error as in #3982 and #3698

**Comment by andrewconnell** [comment]
Can you add a `catch()` share the results of the error returned?

Looping @dzearing from OUIFR as the dialogs are based on the Fabric React dialog component... any known issues with IE11?

Also, you removed the environment question about the environment... assuming SPO since you're using SPFx 1.9.1?

**Comment by henningeiben** [comment]
So I changed my code to 

```typescript
var configSelectionDialog = new CommandSelectionDialog();
configSelectionDialog
.show()
.then(_ => {
  // ...
})
.catch(error => {
  Log.error(LOG_SOURCE, error);
  console.error("something went wrong");
  console.error(error);
});
```
But the `catch` is never reached (the logging is not to be seen, nor adding a breakpoint shows that this code is hit).

This is what I get in the console:
![image](https://user-images.githubusercontent.com/7802924/74472578-74d4e800-4ea2-11ea-8ace-d26b3d0969f4.png)

And: yes, this is concerning SPO - I didn't removed that section intentionally.

**Comment by jonathanAndersonRB** [comment]
I am encountering the same issue, as a workaround in the end I had to add the logic to the dialogue rather than completing once the promise was called.

**Comment by MacBoiler** [comment]
I'm having the same issue with IE11 only...works fine in Chrome.

The webpart in workbench is giving this...

ERROR:
'Proxy' is undefined

CALL STACK:
ReferenceError: 'Proxy' is undefined
   at Anonymous function (https://localhost:4321/dist/engage-scomms-work-request-web-part.js:170070:13)
   at web.get (https://localhost:4321/dist/engage-scomms-work-request-web-part.js:32823:9)
   at FormFields.prototype.componentDidMount (https://localhost:4321/dist/engage-scomms-work-request-web-part.js:36279:9)
   at To (https://spoprod-a.akamaihd.net/files/sp-client/sp-webpart-workbench-assembly_en-au_8e994615ec74add02f49f889aee39279.js:1:1239444)
   at t.unstable_runWithPriority (https://spoprod-a.akamaihd.net/files/sp-client/sp-webpart-workbench-assembly_en-au_8e994615ec74add02f49f889aee39279.js:1:1142572)
   at dl (https://spoprod-a.akamaihd.net/files/sp-client/sp-webpart-workbench-assembly_en-au_8e994615ec74add02f49f889aee39279.js:1:1186417)
   at xo (https://spoprod-a.akamaihd.net/files/sp-client/sp-webpart-workbench-assembly_en-au_8e994615ec74add02f49f889aee39279.js:1:1235451)
   at co (https://spoprod-a.akamaihd.net/files/sp-client/sp-webpart-workbench-assembly_en-au_8e994615ec74add02f49f889aee39279.js:1:1223165)
   at Ho (https://spoprod-a.akamaihd.net/files/sp-client/sp-webpart-workbench-assembly_en-au_8e994615ec74add02f49f889aee39279.js:1:1249724)
   at Qo (https://spoprod-a.akamaihd.net/files/sp-client/sp-webpart-workbench-assembly_en-au_8e994615ec74add02f49f889aee39279.js:1:1249936)

**Comment by andrewconnell** [comment]
Closing this issue... if you continue to have this problem, please submit a new issue with additional context (code, versions, environment, etc)

**Comment by henningeiben** [comment]
Out of curiosity: why is this issue being closed? Was it fixed? I mean ... if it wasn't fixed, why is it closed - and if it wasn't fixed why should I create another issue just like this?

**Comment by andrewconnell** [comment]
I can't explain that... thanks for calling me out as it shouldn't have been closed.

**Comment by IvanTheBearable** [comment]
I can't comment on IE11 but I just started getting this issue with Chrome today after upgrading a web part to SPFx 1.11.

The custom dialog was built base on the sample ColorPickedDialog which includes the following onAfterClose method:

```
protected onAfterClose(): void {
      super.onAfterClose();
      
      // Clean up the element for the next dialog
      ReactDOM.unmountComponentAtNode(this.domElement);
}
```
The ```super.onAfterClose``` call fails silently and never returns. So the ```onAfterClose``` never returns and the ```close``` that called it never completes.

```super.onAfterClose``` looks like this:

```
e.prototype.onAfterClose = function() {
   C.isBaseDialogonAfterCloseKSActive() || l.unmountComponentAtNode(this.domElement)
}
```

I don't know what the killswitch is all about but it hints at known issues in this area. ```isBaseDialogonAfterCloseKSActive``` returns false and ```l.unmountComponentAtNode(this.domElement)``` then fails silently. I tried to drill further into ```unmountComponentAtNode``` but I'm just tracing into the minified JavaScript and that stuff gets pretty hard to follow.

Anyway, the fix seems to be to change the ```onAfterClose`` to this:

```
protected onAfterClose(): void {
      ReactDOM.unmountComponentAtNode(this.domElement);
      super.onAfterClose();
}
```

i.e. unmount **_before_** calling ```super.onAfterClose``` (in fact, the super call doesn't appear to be needed at all). It is strange that the above works since calling ```unmountComponentAtNode``` seems to be all that ```super.onAfterClose``` is doing anyway, but it seems to work.

**Comment by patmill** [comment]
Sorry for going dark.  IE support is deprecated, so we will not be making fixes for it.  @IvanTheBearable - if you are still having problems with the close behaviour - please open a new issue and we will take a look at it.  Thanks.
